### PR TITLE
Update the lookup logic to match the db size lookup logic

### PR DESCRIPTION
**Pull request by KimForss**
## Problem
Fix the lookup logic to match the db lookup logic

## Solution
<Please elaborate the solution for the problem>

## Tests
<Please provide steps to test the PR>

## Notes
<Additional comments for the PR>

### CODE REVIEW

File: deploy/terraform/terraform-units/modules/sap_system/common_infrastructure/variables_local.tf [hunk 1]
```diff
@@ -111,12 +111,19 @@ locals {
...
   default_filepath = local.enable_hdb_deployment ? "${path.module}/../../../../../configs/hdb_sizes.json" : "${path.module}/../../../../../configs/anydb_sizes.json"
+
   sizes            = jsondecode(file(length(var.custom_disk_sizes_filename) > 0 ? var.custom_disk_sizes_filename : local.default_filepath))
-  storage_list     = length(var.databases) > 0 ? lookup(local.sizes, var.databases[0].size).storage : []
-  enable_ultradisk = try(compact([
-    for storage in local.storage_list :
-    storage.disk_type == "UltraSSD_LRS" ? true : ""
-  ])[0], false)
+  db_size_details  = lookup(local.sizes, var.databases[0].size, [])
+  db_sizing        = local.db_size_details != [] ? local.anydb_size_details.storage : []
```
Comment (rukawata) [inline deploy/terraform/terraform-units/modules/sap_system/common_infrastructure/variables_local.tf:117]: `anydb_size_details` or `db_size_details` ?

Comment (rukawata) [reply deploy/terraform/terraform-units/modules/sap_system/common_infrastructure/variables_local.tf:117]: And if it is consistent with app_tier, it should be:

`db_sizing = local. enable_hdb_deployment ? lookup(local.sizes, var.databases[0].size) : []`

Comment (KimForss) [reply deploy/terraform/terraform-units/modules/sap_system/common_infrastructure/variables_local.tf:117]: db_size_details as it is valid for both db backends

File: deploy/terraform/terraform-units/modules/sap_system/common_infrastructure/variables_local.tf [hunk 2]
```diff
@@ -133,6 +127,20 @@ locals {
...
+    compact(
+      [
+        for storage in local.db_sizing : storage.disk_type == "UltraSSD_LRS" ? true : ""
+      ]
+    )[0],
+    false
+  )
+
+
+
```
Comment (rukawata) [inline deploy/terraform/terraform-units/modules/sap_system/common_infrastructure/variables_local.tf:143]: nit:
```suggestion

```

**Review by rukawata — APPROVED** [review]
Thanks for fixing this!

**Pull request merged**